### PR TITLE
Implemented binary_cross_entropy_with_logits in paddle.nn.functional.loss

### LINKED ISSUE #16741: binary_cross_entropy_with_logits

### PR DESCRIPTION
**Pull request by KevinUli**
Close #16741

### CODE REVIEW

**Comment by KevinUli** [comment]
Result from paddle ground truth and other backend differs, so I set `test_values=False`

**Comment by xoiga123** [comment]
@KevinUli How do they differ? If you explicitly set `test_values=False` then you have to implement your own testing from the returned values.

**Comment by KevinUli** [comment]
@xoiga123 When running the function for torch and paddle, 
```
>>> arr = torch.as_tensor([0.5, 0.5,], dtype=torch.float32)
>>> pos_weight = torch.as_tensor([1., 2.,], dtype=torch.float32)
>>> torch.nn.functional.binary_cross_entropy_with_logits(arr, arr, weight=arr, reduction="none", pos_weight=pos_weight)
Tensor([0.3620, 0.4806])
```

```
>>> arr = paddle.to_tensor([0.5, 0.5,], dtype="float32")
>>> pos_weight = paddle.to_tensor([1., 2.,], dtype="float32")
>>> paddle.nn.functional.binary_cross_entropy_with_logits(arr, arr, weight=arr, reduction="none", pos_weight=pos_weight)
Tensor(shape=[2], dtype=float32, place=Place(cpu), stop_gradient=True, [0.36203849, 0.54305774])
```

The pos_weight kwarg seems to work differently for paddle as opposed to other frameworks.

**Comment by xoiga123** [comment]
@KevinUli Yeah seems weird, I'll look into source and see what I can find. Thank you

**Comment by xoiga123** [comment]
@KevinUli Yeah clearly the [pytorch's implementation](https://github.com/pytorch/pytorch/blob/01abccf63faa14b05ffee2474b1fcd7d85d44782/aten/src/ATen/native/Loss.cpp#LL368C12-L368C12) of `pos_weight` is correct, while the [paddle's implementation](https://github.com/PaddlePaddle/Paddle/blob/f0422a28d75f9345fa3b801c01cd0284b3b44be3/python/paddle/nn/functional/loss.py#L831) is incorrect. This seems easily fixable. Please add a comment below the test definition
```python
# TODO: paddle's implementation of pos_weight is wrong
# https://github.com/PaddlePaddle/Paddle/blob/f0422a28d75f9345fa3b801c01cd0284b3b44be3/python/paddle/nn/functional/loss.py#L831
```
And then proceed to open an issue on their side, or open a PR to fix it if you want to. I'll merge this PR but please keep track of when it's fixed and add `pos_weight` back. Thank you :hugs:

**Comment by KevinUli** [comment]
I've opened an issue on their side.

**Review by xoiga123 — APPROVED** [review]
Well done :+1:
